corehq/apps/app_manager/models.py

```diff
@@ -3880,7 +3880,7 @@ def wrap(cls, data):
                 # preserve stubs so couch attachments don't get deleted on save
                 stubs = {}
                 for name, value in list(attachments.items()):
-                    if "stub" in value:
+                    if isinstance(value, dict) and "stub" in value:
```

Comment:
ugh, this is so bad. Took me a while to figure out/remember that `value` is sometimes a string (attachment content) and sometimes a dict. The fact that the value is sometimes a string is very bad because it is so easy to introduce bugs like this. Unfortunately we probably can't fix it the right way (make value always be a dict) because it's an exposed interface in exported apps.